java-checks/src/main/java/org/sonar/java/checks/UndocumentedApiCheck.java

```diff
@@ -198,11 +211,26 @@ private String className() {
     return className;
   }
 
-  private WildcardPattern[] getPatterns() {
-    if (patterns == null) {
-      patterns = PatternUtils.createPatterns(forClasses);
+  private WildcardPattern[] getInclusionPatterns() {
+    if (inclusionPatterns == null) {
+      if (StringUtils.isEmpty(forClasses)) {
+        forClasses = "**";
+      }
+
+      inclusionPatterns = PatternUtils.createPatterns(forClasses);
+    }
+    return inclusionPatterns;
+  }
+
+  private WildcardPattern[] getExclusionPatterns() {
+    if (exclusionPatterns == null) {
+      if (StringUtils.isEmpty(exclusion)) {
+        return new WildcardPattern[0];
```

Comment:
Well, as you are not setting the `exclusionPatterns` field, but directly returning in case of empty `exclusion` param, it will remains `null`.